pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -631,6 +639,16 @@ public boolean useSpools(Map<String, String> options) {
     return Boolean.parseBoolean(optionValue);
   }
 
+  /**
+   * Helper method to get stack trace as string for logging
+   */
+  private static String getStackTraceString(Throwable t) {
```

Comment:
this could be costly at higher qps. also I think we should leverage some utils class for this.